### ISSUE TITLE
Need support for new type of device ID

### ISSUE THREAD

**Issue by vishwanathan-iyer** [jayantsolanki/IoT-Automation-V2#2]
**Is your feature request related to a problem? Please describe.**
Keeping security in mind, IoT development boards have a security element where device IDs like 0123351C540EABCDEF are stored by default from the factory.  

**Describe the solution you'd like**
Add support for device IDs apart from mac  ID.

**Describe alternatives you've considered**
Can configure dummy macid but, I want to take advantage of this security feature.

**Additional context**
This  ID is stored in hardware on board to which the microcontroller talks and sends it over WIFI.

**Comment by jayantsolanki** [comment]
For the ESP devices, I found this on their website: https://bbs.espressif.com/viewtopic.php?t=1303
tl;dr, ESP doesnt have unique serial numbers/ids, other than macids, what I can do is to generate unique 'uuid' for each ESP/other devices, this can help in cataloging them and uniquely identify them. However we will have to keep track of uuid, just in case for safekeeping.
More on uuid: https://en.wikipedia.org/wiki/Universally_unique_identifier
